### PR TITLE
add severity-filter feature

### PR DESCRIPTION
**Pull request by ansidorov**
Resolve https://github.com/ajinabraham/njsscan/issues/69

Running scan without `.njsscan` file:
```
❯ docker run -v $(pwd):/app njsscan /app --json | jq '.nodejs[].metadata.severity'
"INFO"
"INFO"
"WARNING"
"INFO"
"INFO"
"WARNING"
"WARNING"
"WARNING"
"WARNING"
"ERROR"
"ERROR"
"ERROR"
"WARNING"
"ERROR"
```

Add `.njsscan` file with content:
```
---
- severity-filter:
  - ERROR
  - WARNING
```

And got only `WARNING` and `ERROR` results:
```
❯ docker run -v $(pwd):/app njsscan /app --json | jq '.nodejs[].metadata.severity'
"WARNING"
"WARNING"
"WARNING"
"WARNING"
"WARNING"
"ERROR"
"ERROR"
"ERROR"
"WARNING"
"ERROR"
```

### CODE REVIEW

File: njsscan/njsscan.py [hunk 1]
```diff
@@ -101,6 +103,17 @@ def post_ignore_rules(self):
             if rule_id in self.result['templates']:
                 del self.result['templates'][rule_id]
 
+    def post_ignore_rules_by_severity(self):
```
Comment (ajinabraham) [inline njsscan/njsscan.py:106]: This method has to run for both `nodejs` and `templates` keys. We have results from both semgrep and pattern matcher and we handle it separately. I suggest calling the `post_ignore_rules_by_severity() method twice with `nodejs` and `template` as result keys.

Something like 

```python
def post_ignore_rules_by_severity(self, key):
        """Filter findings by rule severity."""
         del_keys = set()
         for rule_id, details in self.result[key].items():
             issue_severity = details.get('metadata').get('severity')
             if issue_severity not in self.options['severity_filter']:
                 del_keys.add(rule_id)
         for rid in del_keys:
             if rid in self.result[key]:
                 del self.result[key][rid]
 ```

**Review by ajinabraham — CHANGES_REQUESTED** [review]

File: njsscan/njsscan.py (hunk 1 above)
Comment (ansidorov) [reply njsscan/njsscan.py]: Corrected, did I hear you right?

Comment (ajinabraham) [reply njsscan/njsscan.py]: Perfect!

**Review by ajinabraham — APPROVED** [review]
LGTM 👍

**Pull request merged**